src/shared/proto/messages_robocup_ssl_detection_tracked.proto

```diff
@@ -0,0 +1,122 @@
+syntax = "proto2";
+
+message Vector2 {
+    required float x = 1;
+    required float y = 2;
+}
+
+message Vector3 {
+    required float x = 1;
+    required float y = 2;
+    required float z = 3;
+}
+
+// The team color of the robot
+enum TeamColor {
+    UNKNOWN = 1;
+    YELLOW = 2;
+    BLUE = 3;
+}
+
+message RobotId {
+    // The robot number
+    required uint32 id = 1;
+    // The team color
+    required TeamColor team_color = 2;
+}
+
+enum ChipKickDetectionSupport {
+    UNSUPPORTED = 0;
+    FIRST_TOUCHDOWN_ONLY = 1;
+    ALL_TOUCHDOWNS = 2;
+}
+
+// A single tracked ball
+message TrackedBall {
+    // The quality of the detection
+    // A value between 0 (not visible) and 1 (visible)
+    // The exact implementation depends on the source software
+    optional float quality = 1;
+
+    // The position (x, y, height) [m] in the ssl-vision coordinate system
+    required Vector3 pos = 2;
+
+    // The velocity [m/s] in the ssl-vision coordinate system
+    optional Vector3 vel = 3;
+
+    // The level of chip kick detection support by the source implementation
+    required ChipKickDetectionSupport chip_kick_detection_support = 4;
+
+    // The touch down location(s), if the ball is flying.
+    // Depending on the source implementation, the next or all touch down locations may be provided
+    // ChipKickDetectionSupport#UNSUPPORTED -> there are no valid touch down locations
+    // ChipKickDetectionSupport#FIRST_TOUCHDOWN_ONLY -> there is only one touch down location, although the ball may have more
+    // ChipKickDetectionSupport#ALL_TOUCHDOWNS -> all touch down locations are included
+    repeated Vector2 touch_down_locations = 5;
```

Comment:
I would recommend to remove this field completely. The first touchdown location is implicit in the ball position and velocity as it fully describes the ball state to compute a touchdown.
Further touchdown locations have very low confidence. Bounce behaviour depends a lot on (not observable) ball spin. It also differes slightly depending on touchdown location on the field (due to construction differences).
I think this field is rather confusing for the intended users (new teams) and would drop it completely.

Reply:
agreed